bindings/experimental/distrdf/python/DistRDF/HeadNode.py

```diff
@@ -112,30 +197,21 @@ def get_num_entries(self):
             int: This is the computed number of entries in the input dataset.
 
         """
-        first_arg = self.args[0]
-        if isinstance(first_arg, int):
-            # If there's only one argument
-            # which is an integer, return it.
-            return first_arg
-        elif isinstance(first_arg, ROOT.TTree):
+        # First argument can be a string or a TTree/TChain
+        firstarg = self.args[0]
+        if isinstance(firstarg, ROOT.TTree):
             # If the argument is a TTree or TChain,
             # get the number of entries from it.
-            return first_arg.GetEntries()
+            return firstarg.GetEntries()
+        elif isinstance(firstarg, str):
+            # Construct a ROOT.TChain object with the name of the tree
+            chain = ROOT.TChain(firstarg)
 
-        second_arg = self.args[1]
+            # Add the list of input files to the chain
+            for fname in self.inputfiles:
+                chain.Add(fname)
 
-        # Construct a ROOT.TChain object
-        chain = ROOT.TChain(first_arg)
-
-        if isinstance(second_arg, str):
-            # If the second argument is a string
-            chain.Add(second_arg)
-        else:
-            # If the second argument is a list or vector
-            for fname in second_arg:
-                chain.Add(str(fname))  # Possible bug in conversion of string
-
-        return chain.GetEntries()
+            return chain.GetEntries()
```

Comment:
this is super costly to do at every call, please add a TODO/FIXME

Reply:
Done in latest commit